auth/token_authentication.py

```diff
@@ -62,6 +45,50 @@ def authenticate_credentials(token: str) -> AppUser:
             "Your Gooey.AI account has been disabled for violating our Terms of Service. "
             "Contact us at support@gooey.ai if you think this is a mistake."
         )
-        raise HTTPException(status_code=401, detail={"error": msg})
+        raise AuthenticationError(msg)
 
     return user
+
+
+class APIAuth(SecurityBase):
```

Comment:
I created a new class instead of using FastAPI's default `HTTPBearer` because it has terrible error messages and uses 403 for all auth problems.

https://github.com/fastapi/fastapi/blob/0.85.0/fastapi/security/http.py#L113-L133

My changes are compatible with what we already do - nothing has changed behaviourally for the API.